### ISSUE TITLE
Sort Month when Using Month as the Grouper

### ISSUE THREAD

**Issue by Yun-Wu** [wildlife-dynamics/ecoscope-workflows#235]

**Comment by atmorling** [comment]
#203 added a `sort_values` task - would it be reasonable to use that as step before applying the temporal index?

I'm asking because I was tempted to add a `sort` param to `create_map_layer` but figured there's going to be a bunch of places in a workflow (such as `add_temporal_index`) where we might want to optionally sort, so I figure it makes more sense to have a generic task.

**Comment by Yun-Wu** [comment]
@atmorling Yeah, you're right - we definitely need a generic sorting task. The month sorting is a bit tricky since we can't just sort it alphabetically. I'm kinda torn on where to put it though. We could either:

1. Put it into the general sort function as a special case, or
2. Put it inside add_temporal_index and only sort when we're using 'month' as the grouper.

What's your take on this? Any preference?

**Comment by atmorling** [comment]
You're right, that is tricky :exploding_head: 
I think 2 is easier, if we were going to do something like 1, a separate task altogether might make sense. ie `sort_dates` or even `categorical_sort`
